db.go

```diff
@@ -528,6 +528,34 @@ func (db *DB) Close() error {
 	return merr.Err()
 }
 
+// DisableCompactions disables compactions.
+func (db *DB) DisableCompactions() error {
+	db.stopc <- struct{}{} // TODO: Can this block?
```

Comment:
I suppose you want to cancel compactions here? The problem is that this may cancel any background processing forever https://github.com/prometheus/tsdb/blob/master/db.go#L272.

I would say just leave it out for now. That may cause the call to wait for quite a bit if there are lots of compactions in progress. But that's rare. We can figure out a separate approach for interrupting ongoing compactions later.

There are a few concerns here. For example, do we want to terminate and cleanup in-progress compactions, or just prevent that a single compaction trigger to process multiple compactions. We can worry about them later.